feat(toolbar): Add setup instructions for the Dev Toolbar using the react hook useSentryToolbar (#12845)
getsentry/sentry-docs: docs/product/dev-toolbar/faq.mdx

description: "Frequently asked questions about the Dev Toolbar."
---

> <Expandable title="In what environments should I enable the Dev Toolbar?">

Reviewer: I had suggested to name the npm package `developer toolbar` (I don't have strong feels about it), but if "Dev Toolbar" is the official product name for it, we should make the npm package name match.

(why can't this be "Sentry Toolbar", does it need dev at all?)

Author: renaming the thing makes sense. the repo is called `sentry-toolbar` already.

I think using `development`, especially in the npm package name, was meant to make it not sound like an environment or a dev/alpha/beta tag. A proper rename avoids any confusion in that area.

Author: Put up a rename PR: https://github.com/getsentry/sentry-toolbar/pull/223

The current state is that the npm publish of `@sentry/developer-toolbar` (the old name) isn't complete yet, there's a bug in the pipeline.

I've asked for help with that pipeline bug, so if it's not disruptive to debugging we can land the rename name. I'll check in on that. Otherwise I might just keep working at that pipeline problem first which means we will publish the old name once. After that's working we can land the rename.

Since the Dev Toolbar will be visible to users within your app, it's important to consider which environments should render it.

> If your web application does require authentication to access:

Reviewer: ```suggestion
If your web application requires authentication to access:
```

> - In dev and staging, always initialized the Dev Toolbar.

Reviewer: Should we write out `development` irt env, as to not conflate with the "dev" in "Dev Toolbar"?

```suggestion
- In development and staging, always initialized the Dev Toolbar.
```

- In production conditionally initialize the Dev Toolbar when an employee is logged in.

If you web application does not require authenticaion:
- In dev and staging environments initialize the Toolbar at all times.
- In production environments, do not initialize the Toolbar.

> Initializing the Dev Toolbar allows all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.

Reviewer: ```suggestion
Initializing the Dev Toolbar allows all developers and testers to use it and quickly go from the page they're looking at, back to Sentry for further debugging.
```

> In production it can make it easier for developers to reproduce issues, but it should not be initialized for all users of the site -- only when an employee/engineer/etc visits.

Reviewer: Feels like this warrants an alert-style component
